random_numbers.c

```diff
@@ -0,0 +1,55 @@
+/* The following code offers an alterate pseudo random number generator
+   namely XORSHIFT128+ to use instead of C's rand(). Its performance is 
+   on par with C's rand().
+ */
+
+#include <stdint.h>
+#include "n2n.h"
+#include "random_numbers.h"
+
+struct rn_generator_state_t {
+    uint64_t a, b;
+ };
+
+/* The state must be seeded in a way that it is not all zero, choose some
+   arbitrary defaults (in this case: taken from splitmix64)
+ */
+static struct rn_generator_state_t rn_current_state
+			       = { .a    = 0x9E3779B97F4A7C15,
+			           .b    = 0xBF58476D1CE4E5B9
+};
+
+int32_t n2n_srand (uint64_t seed) {
+    /* apply splitmix64 algorithm (found on same wiki page as xorshift)
+       twice on the SEED to calcualte A and B of XORSHIFT128PLUS' state */
+    rn_current_state.a = seed;
```

Comment:
This seems different from the wikipedia implementation. This code applies:

```
rn_current_state.a = splitmix(seed)
rn_current_state.b = splitmix(seed)
```

whereas the wikipedia code runs two rounds of splitmix :
```
state = {seed}
rn_current_state.a = splitmix(&state)
rn_current_state.b = splitmix(&state)
```
Moreover in `rn_current_state.a = seed` the contant `0x9E3779B97F4A7C15` is not applied. The wikipedia implementation can be also found in other online source written in go:

https://github.com/vpxyz/xorshift/blob/master/xorshift128plus/xorshift128plus.go#L27
https://github.com/vpxyz/xorshift/blob/master/internal/internal.go#L30

Reply:
The addition only gets applied to the state which is _not_ used further in that specific call again - but in the next. For every seeding according to the given pattern, splitmix algorithm is  applied twice in a row only - with initial state of {0}. 

That way, implementation is more linear, i.e. contains less sub routine calls, and there is no need to keep track of a static splitmix state. Isn't the result the same? What did I miss?

If you are not convinced, let me know, I definitely can change it.

Reply:
In a test run, I implemented the wiki-version which delivers the **very same results** as the streamlined implementation above. I personally think that the wiki-implementation kind of clutters the code.

I will upload the wiki-solution in a new commit. Please feel free to choose and incorporate any of the two commits/versions into the dev branch.

Reply:
Done.